### PR TITLE
Adding StaticTransformListener in Python

### LINKED ISSUE #662: Create a StaticTransformListener
#### Feature description
The `TransformListener` in `tf2_ros` subscribes to both the `/tf` and `/tf_static` topics and provides a unified transform tree. However, there are many cases where a node will _only_ be using transforms that are known to be static, e.g., looking up transforms from a sensor frame to the body frame. In those cases, all of the overhead in maintaining the high-rate transform data as provided by the `/tf` topic is totally wasted. Given that many nodes use `TransformListener`, this can quickly add up to a lot of wasted resources in systems with large numbers of nodes.

I propose that we implement a `StaticTransformListener` that _only_ subscribes to `/tf_static`.

#### Implementation considerations
We could define a base class for both `TransformListener` and `StaticTransformListener` so keep the interface consistent between the two. The aim of the design would be to not change anything about `TransformListener` itself from a user perspective.

If that is not possible, then if we could at least make the `init` method protected [instead of private](https://github.com/ros2/geometry2/blob/5eabd34b3cc7a1be9e413e7b87093f63f3a2c020/tf2_ros/include/tf2_ros/transform_listener.h#L157), it will let users override this behaviour themselves.

### PR DESCRIPTION
**Pull request by CursedRock17**
As per the title, this pull request is meant to compliment #673 and hopefully close #662 by adding the `StaticTransformListener` in Python.

### CODE REVIEW

File: tf2_ros_py/tf2_ros/static_transform_listener.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
+# Copyright (c) 2008, Willow Garage, Inc.
```
Comment (ahcorde) [inline tf2_ros_py/tf2_ros/static_transform_listener.py]:
```suggestion
# Copyright (c) 2024, Open Source Robotics Foundation, Inc.
```

File: tf2_ros_py/tf2_ros/static_transform_listener.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+# author: Wim Meeussen
```
Comment (ahcorde) [inline tf2_ros_py/tf2_ros/static_transform_listener.py]: remove or add yourself

File: tf2_ros_py/tf2_ros/static_transform_listener.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        *,
+        spin_thread: bool = False,
+        static_qos: Optional[Union[QoSProfile, int]] = None,
+        tf_static_topic: str = DEFAULT_STATIC_TF_TOPIC
+    ) -> None:
+        """
+        Constructor.
+
+        :param buffer: The buffer to propagate changes to when tf info updates.
+        :param node: The ROS2 node.
```
Comment (ahcorde) [inline tf2_ros_py/tf2_ros/static_transform_listener.py]:
```suggestion
        :param node: The ROS 2 node.
```

**Review by ahcorde — CHANGES_REQUESTED** [review]
can you also add the test ?

**Comment by CursedRock17** [comment]
> can you also add the test ?

As in a whole new testing suite or another set of individualized tests? The `StaticTransformListener` should be applicable in `test_static_broadcaster_and_listener`, just a faster version of the `TransformListener`.

**Comment by ahcorde** [comment]
> > can you also add the test ?
> 
> As in a whole new testing suite or another set of individualized tests? The `StaticTransformListener` should be applicable in `test_static_broadcaster_and_listener`, just a faster version of the `TransformListener`.

The mentioned PR adds some new tests to `test_transform_listener.cpp`, I would suggest to add something similar to `tf2_ros_py`

**Comment by ahcorde** [comment]
Pulls: ros2/geometry2#719
Gist: https://gist.githubusercontent.com/ahcorde/8eca404e13a04ad871146a39db4cdcc2/raw/9d524ee1ccc2723415616719d3026e29857f0835/ros2.repos
BUILD args:   --packages-above-and-dependencies tf2_ros_py --packages-above-and-dependencies tf2_ros_py
TEST args:  --packages-above tf2_ros_py --packages-above tf2_ros_py
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/14794
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/22099/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/16355/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/1619/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/22839/)

**Review by ahcorde — APPROVED** [review]

**Pull request merged**